mllib/src/main/scala/org/apache/spark/ml/regression/GeneralizedLinearRegression.scala

```diff
@@ -761,16 +783,24 @@ class GeneralizedLinearRegressionModel private[ml] (
    * otherwise generates a new column and sets it as the prediction column on a new copy
    * of the current model.
    */
-  private[regression] def findSummaryModelAndPredictionCol()
-    : (GeneralizedLinearRegressionModel, String) = {
+  private[regression] def getSummaryModel: GeneralizedLinearRegressionModel = {
     $(predictionCol) match {
       case "" =>
         val predictionColName = "prediction_" + java.util.UUID.randomUUID.toString
-        (copy(ParamMap.empty).setPredictionCol(predictionColName), predictionColName)
-      case p => (this, p)
+        copy(ParamMap.empty).setPredictionCol(predictionColName)
+      case p => this
```

Comment:
That sounds good.  That way, we can just use the model internally in the summary and will not need to copy the model parameters below.  I'll do that.